### PR TITLE
Add Single-Quote Checker to marking logic

### PR DESCRIPTION
**Pull request by cissyxyu**
## WHAT
Add check to quill marking logic that catches when students use two adjacent single quotes in a response, instead of a double quote.

## WHY
We want to warn students about this error as it commonly trips up students because they can't visually distinguish two single quotes and a double quote.

## HOW
Add the catcher in marking logic, and update the npm package and package references accordingly.

### Screenshots
Screenshot 2023-12-14 at 6 46 12 PM
Screenshot 2023-12-14 at 6 57 25 PM
Screenshot 2023-12-14 at 6 57 28 PM

### Notion Card Links
[(Please provide links to any relevant Notion card(s) relevant to this PR.)](https://www.notion.so/quill/P3-Fix-quotation-mark-grading-in-Connect-and-Grammar-751cb7e21e6a44feb5e2a99aa92c2d08?pvs=4)

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  YES
Have you deployed to Staging? | Not yet - deploying now!
Self-Review: Have you done an initial self-review of the code below on Github? | Yes
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | Yes

### CODE REVIEW

File: packages/quill-marking-logic/src/libs/matchers/quotation_mark_match.spec.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import { assert } from 'chai';
+
+import { quotationMarkChecker, quotationMarkMatch } from './quotation_mark_match';
+
+import {Response,PartialResponse} from '../../interfaces'
+import {feedbackStrings} from '../constants/feedback_strings'
+import {conceptResultTemplate} from '../helpers/concept_result_template'
+
+describe('The quotationMarkMatch function', () => {
+
```
Comment (brendanshean) [inline packages/quill-marking-logic/src/libs/matchers/quotation_mark_match.spec.ts:10]: What about extracting out the three savedReponses here:
```
  const savedResponses: Array<Response> = [
    ...
  ];
```

**Review by brendanshean — APPROVED** [review]
Nice work on getting this marking logic implemented.

**Review by emilia-friedberg — APPROVED** [review]
LGTM. It looks like the spec calls for some changes in the admin tools too, are those coming in a follow-up PR?

**Comment by cissyxyu** [comment]
> LGTM. It looks like the spec calls for some changes in the admin tools too, are those coming in a follow-up PR?

The admin tool changes are also in this PR -- it's actually just adding to two lists of constants. Since the badge and button functionality has already been built, this just adds one more badge/button to the existing UI.

**Pull request merged**